tests/app/inbound_number/test_rest.py

```diff
@@ -47,14 +47,36 @@ def test_gets_inbound_numbers(self, admin_request, mocker):
 
 class TestSetInboundNumberOff:
     def test_sets_inbound_number_active_flag_off(self, admin_request, mocker):
-        dao_set_inbound_number_active_flag = mocker.patch('app.inbound_number.rest.dao_set_inbound_number_active_flag')
+        """
+        Test that the endpoint sets the active flag of an inbound number to False.
+        """
+        dao_set_inbound_number_active_flag = mocker.patch(
+            'app.inbound_number.rest.dao_set_inbound_number_active_flag', return_value=None
+        )
 
         inbound_number_id = uuid.uuid4()
         admin_request.post(
             'inbound_number.post_set_inbound_number_off', _expected_status=204, inbound_number_id=inbound_number_id
         )
         dao_set_inbound_number_active_flag.assert_called_with(inbound_number_id, active=False)
 
+    def test_returns_400_when_inbound_number_does_not_exist(self, admin_request, mocker):
+        """
+        Test that the endpoint returns a 400 error when the inbound number does not exist.
+        """
+        error_message = 'Inbound number with id {} does not exist'
+        mocker.patch(
+            'app.inbound_number.rest.dao_set_inbound_number_active_flag', side_effect=ValueError(error_message)
+        )
+
+        non_existent_id = uuid.uuid4()
+        response = admin_request.post(
+            'inbound_number.post_set_inbound_number_off', _expected_status=400, inbound_number_id=non_existent_id
+        )
+
+        assert response['errors'][0]['error'] == 'BadRequestError'
+        assert f'Inbound number with id {non_existent_id} does not exist' in response['errors'][0]['message']
```

Comment:
This test also should not mock anything.  It should not create an inbound number and then call the endpoint with a random UUID in the URL.  Test that the response is 400.